### PR TITLE
Feat/prevent reinit

### PR DESCRIPTION
**Pull request by Ayoazeez26**
- [x] issue #47
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)

Added a check to see if canvas has been initialised in the database

### CODE REVIEW

File: backend/routes/canvas.go [hunk 1]
```diff
@@ -14,22 +14,29 @@ func InitCanvasRoutes() {
...
 
 func initCanvas(w http.ResponseWriter, r *http.Request) {
-	// TODO: Check if canvas already exists
-	totalBitSize := core.ArtPeaceBackend.CanvasConfig.Canvas.Width * core.ArtPeaceBackend.CanvasConfig.Canvas.Height * core.ArtPeaceBackend.CanvasConfig.ColorsBitWidth
-	totalByteSize := (totalBitSize / 8)
-	if totalBitSize%8 != 0 {
-		// Round up to nearest byte
-		totalByteSize += 1
+	// Check if the 'canvas' property exists in core.ArtPeaceBackend.CanvasConfig
+	if core.ArtPeaceBackend.CanvasConfig != nil && core.ArtPeaceBackend.CanvasConfig.Canvas != nil {
```
Comment (b-j-roberts) [inline backend/routes/canvas.go:18]: This check is good, but I was thinking of having another check like this `core.ArtPeaceBackend.Databases.Redis.Exists(context.Background(), "canvas").Val() == 1` to make sure the canvas does not already exist in Redis's DB.

Comment (Ayoazeez26) [reply backend/routes/canvas.go:18]: Alright, thank you for the feedback.
Adding the check now

**Comment by b-j-roberts** [comment]
Nice work!

For the future, you can use `go build` or `go run main` inside the `backend` directory to test building and/or running the go code.

**Comment by Ayoazeez26** [comment]
I'll keep that in mind, thanks.

**Pull request merged**